messaging/src/main/java/org/axonframework/tracing/NestingSpanFactory.java

```diff
@@ -0,0 +1,176 @@
+        if (!isChildTrace && messageIsWithinTimeLimit(parentMessage)) {
+            return delegateSpanFactory.createHandlerSpan(operationNameSupplier, parentMessage, true, linkedParents);
+        }
+        return delegateSpanFactory.createHandlerSpan(operationNameSupplier, parentMessage, isChildTrace, linkedParents);
+    }
+
+    private boolean messageIsWithinTimeLimit(Message<?> parentMessage) {
+        if (!(parentMessage instanceof EventMessage)) {
+            return true;
+        }
+        Instant timestamp = ((EventMessage<?>) parentMessage).getTimestamp();
+        return clock.instant().isBefore(timestamp.plus(timeLimit));
+    }
+
+    @Override
+    public Span createDispatchSpan(Supplier<String> operationNameSupplier, Message<?> parentMessage,
+                                   Message<?>... linkedSiblings) {
+        return delegateSpanFactory.createDispatchSpan(operationNameSupplier, parentMessage, linkedSiblings);
+    }
+
+    @Override
+    public Span createInternalSpan(Supplier<String> operationNameSupplier) {
+        return delegateSpanFactory.createInternalSpan(operationNameSupplier);
+    }
+
+    @Override
+    public Span createInternalSpan(Supplier<String> operationNameSupplier, Message<?> message) {
+        return delegateSpanFactory.createInternalSpan(operationNameSupplier, message);
+    }
+
+    @Override
+    public void registerSpanAttributeProvider(SpanAttributesProvider provider) {
+        delegateSpanFactory.registerSpanAttributeProvider(provider);
+    }
+
+    @Override
+    public <M extends Message<?>> M propagateContext(M message) {
+        return delegateSpanFactory.propagateContext(message);
+    }
+
+
+    /**
+     * Creates a builder that will create a {@link NestingSpanFactory}.
+     * <p>
+     * Requires the delegate {@link SpanFactory} to be configured.
+     * <p>
+     * The {@link Clock} defaults to the system utc time and the timeLimit {@link Duration} defaults to two minutes.
+     */
+    public static class Builder {
+
+        private SpanFactory delegateSpanFactory;
+        private Duration timeLimit = Duration.ofMinutes(2);
+        private Clock clock = Clock.systemUTC();
+
+        /**
+         * Defines the delegate {@link SpanFactory} to use, which actually provides the spans.
+         *
+         * @param spanFactory The {@link SpanFactory} to configure for use.
+         * @return The current Builder instance, for fluent interfacing.
+         */
+        public Builder delegate(SpanFactory spanFactory) {
+            BuilderUtils.assertNonNull(spanFactory, "The spanFactory should not be null");
+            this.delegateSpanFactory = spanFactory;
+            return this;
+        }
+
+        /**
+         * Configures the {@link Duration} since original publishing of an event during which it should be considered a
+         * nested span. After that duration, it will become its own separate trace.
```

Comment:
With similar reasoning as the previous default description, I'd expect these methods to share it too. At least from my personal approach, I let the IDE open the documentation for *a method* I'm looking at, not the entire class. Seeing the default value here (too) is thus pretty valuable.